### ISSUE TITLE
Missing checks for sending assets to non existent dNFTs

### ISSUE THREAD

**Issue by zobront** [DyadStablecoin/contracts-v3#5]
### Summary

There are two places in the code where the user inputs the `id` of the dNFT they would like to send assets to.

- In `deposit()`, the user sends `msg.value` directly to the dNFT
- In `move()`, the user sends their own shares to another dNFT

In both cases, there are no checks to ensure that the receiving dNFT actually exists.

This could allow users to deposit assets to a non-existent ID that would become permanently trapped in the contract.

### Proof of Concept

```solidity
function test_CanDepositToNonExistantId() public {
    dNft.deposit{value: 5000 ether}(987654321);
}
```

### Recommendation

Add a check in both `deposit()` and `move()` to ensure that the ID exists:

```diff
function deposit(uint id) external payable returns (uint) {
+   require(id < totalSupply(), "dNFT does not exist");
    return _deposit(id);
 }
```

**Comment by SharifElfouly** [comment]
fixed by https://github.com/DyadStablecoin/contracts-v3/pull/11

**Comment by zobront** [comment]
Fix confirmed in #11.
